playlist.c

```diff
@@ -2576,6 +2735,39 @@ playlist_t *playlist_init(const playlist_config_t *config)
    /* Attempt to read any existing playlist file */
    playlist_read_file(playlist);
 
+   /* try use relative paths if enabled */   
+   if (use_relative_path)
+   {
+      for (i = 0; i < playlist-> size; i++)
+      {
+         struct playlist_entry *entry = &playlist->entries[i];
+         if (!string_is_empty(entry->relative_path))
+         {
+            if (entry->path)
+               free(entry->path);
+            tmp_entry_path[0] = '\0';
+            path_resolve_to_local_file_system(tmp_entry_path, entry->relative_path, config->base_content_directory, PATH_MAX_LENGTH);
```

Comment:
Just a little stylistic thing - use `sizeof(tmp_entry_path)` instead of `PATH_MAX_LENGTH` here (and below, as well)